lib/fetch/response.js

```diff
@@ -467,7 +467,7 @@ function initializeResponse (response, init, body) {
 
   // 5. If init["headers"] exists, then fill response’s headers with init["headers"].
   if ('headers' in init && init.headers != null) {
-    fill(response[kState].headersList, init.headers)
+    fill(response[kHeaders], init.headers)
```

Comment:
I think we've had some discussion about this line before, using headersList here was always incorrect, but a workaround for some bugs with data urls. However it'd also bypass header name/header value validation which *almost* sounds like a vulnerability...